### PR TITLE
motives

### PR DESCRIPTION
**Pull request by szaxar**

### CODE REVIEW

File: src/main/java/monitoringsystemturbo/controller/MainController.java [hunk 1]
```diff
@@ -38,6 +38,10 @@ public void showMainWindow(List<Application> loadedApplications) throws IOExcept
         ExportController exportController = new ExportController(primaryStage);
         mainPresenter.setExportController(exportController);
 
+        MotivesController motivesController=new MotivesController(primaryStage);
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/controller/MainController.java]: pls format file, ctrl alt L in intellij

File: src/main/java/monitoringsystemturbo/controller/MotivesController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        FXMLLoader loader = new FXMLLoader();
+        loader.setLocation(this.getClass().getResource("/motivesView.fxml"));
+        Parent rootLayout = loader.load();
+
+        Stage dialogStage = new Stage();
+        dialogStage.setTitle("Motives");
+        dialogStage.initModality(Modality.WINDOW_MODAL);
+        dialogStage.initOwner(primaryStage);
+
+
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/controller/MotivesController.java]: too many empty lines, pls focus better on formatting, it is important to guide consistent style

File: src/main/java/monitoringsystemturbo/presenter/ConfirmExportPresenter.java [hunk 3]
```diff
@@ -62,13 +65,13 @@ public void setPrimaryStage(Stage primaryStage) {
 
     private void setCheckboxes() {
         boolean areAllCheckboxesSelected = areAllCheckboxesSelected();
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/presenter/ConfirmExportPresenter.java]: there is no need to use this variable. It has the same name as a method, so it just an unnecessary line

File: src/main/java/monitoringsystemturbo/presenter/ConfirmExportPresenter.java [hunk 4]
```diff
@@ -62,13 +65,13 @@ public void setPrimaryStage(Stage primaryStage) {
...
-        for (CheckBox box : checkBoxes) {
+        for (JFXCheckBox box : checkBoxes) {
             box.setSelected(!areAllCheckboxesSelected);
         }
     }
 
     private boolean areAllCheckboxesSelected() {
-        for (CheckBox box : checkBoxes) {
+        for (JFXCheckBox box : checkBoxes) {
             if (!box.isSelected()) {
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/presenter/ConfirmExportPresenter.java:76]: you don't need to use if, just return what's inside if body

Comment (mhabzda) [reply src/main/java/monitoringsystemturbo/presenter/ConfirmExportPresenter.java:76]: I know that's not your change right now, but let's follow The Boy Scout Rule and improve code when seeing some smells

File: src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java [hunk 5]
```diff
@@ -1,107 +1,178 @@
...
+    @FXML
+    public void initialize() {
+        fromDatePicker.setValue(now());
+        toDatePicker.setValue(now());
+        fromTimePicker.setValue(LocalTime.of(0, 0));
+        toTimePicker.setValue(LocalTime.now());
+    }
+
+    @FXML
+    public void onConfirm() {
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java:67]: this is a really long method, consider splitting it into reasonable chunks(other methods) with good namings

File: src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java [hunk 6]
```diff
@@ -1,107 +1,178 @@
...
+                List<String> applicationsToExport = confirmExportController.showConfirmationAndGetAppList(trackingService.getApplicationsNames());
+                if (!confirmExportController.getCancelValue()) {
+                    if (wholeRangeCheckBox.isSelected()) {
+                        mainExporter.export(trackingService, applicationsToExport);
+                    } else {
+                        mainExporter.export(trackingService, applicationsToExport, fromTime, toTime);
+                    }
+                    Alert successAlert = new Alert(Alert.AlertType.INFORMATION);
+                    successAlert.setTitle("Success!");
+                    successAlert.setHeaderText(null);
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java]: especially you can make a method for alert displaying, it almost the same code all over the place just with different arguments

File: src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java [hunk 7]
```diff
@@ -1,107 +1,178 @@
...
+            if (!fromBeginCheckBox.isSelected()) wholeRangeCheckBox.setDisable(false);
+        }
+    }
+
+    @FXML
+    public void onFromBegin() {
+        if (fromBeginCheckBox.isSelected()) {
+            fromDatePicker.setValue(LocalDate.of(1970, 1, 1));
+            fromTimePicker.setValue(LocalTime.of(0, 0));
+            fromDatePicker.setEditable(false);
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java]: maybe there is a way to get this lines together and make a good method for that? It is used very often in this file, so if there is a possibility to get some lines together and extract it to method pls do that

File: src/main/java/monitoringsystemturbo/presenter/MainPresenter.java [hunk 8]
```diff
@@ -305,4 +308,18 @@ public void setApplicationListController(ApplicationListController applicationLi
...
+    public void setMotivesController(MotivesController motivesController) {
+        this.motivesController = motivesController;
+    }
+
+    @FXML
+    public void onMotives() {
+        try {
+            motivesController.showMotivesView();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/presenter/MainPresenter.java]: Are you sure we don't want to display some message to a user, but just print a stack trace? If it is an important error which could cause an unexpected app behavior consider notifying user about that

File: src/main/java/monitoringsystemturbo/presenter/MotivesPresenter.java [hunk 9]
```diff
@@ -0,0 +1,4 @@
+package monitoringsystemturbo.presenter;
+
+public class MotivesPresenter {
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/presenter/MotivesPresenter.java]: why is this class empty? If it not necessary pls remove

**Comment by mhabzda** [comment]
and resolve conflicts pls

File: src/main/java/monitoringsystemturbo/controller/ApplicationListController.java [hunk 10]
```diff
@@ -37,7 +36,7 @@ public Application showAddView() throws IOException {
 
         addApplicationPresenter = loader.getController();
         addApplicationPresenter.setPrimaryStage(dialogStage);
-
+        addApplicationPresenter.reflesh();
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/controller/ApplicationListController.java:39]: shouldn't it be refresh?

File: src/main/java/monitoringsystemturbo/controller/MotivesController.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+public class MotivesController {
+
+    private Stage primaryStage;
+
+    public MotivesController(Stage primaryStage) {
+        this.primaryStage = primaryStage;
+    }
+
+
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/controller/MotivesController.java]: delete this new line

File: src/main/java/monitoringsystemturbo/presenter/MotivesPresenter.java [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+    @FXML
+    public void initialize() {
+        textColorPicker.setValue(textCollor);
+        controllersColorPicker.setValue(controllerColor);
+        backgroundColorPicker.setValue(backgroundColor);
+        ripplerColorPicker.setValue(ripplerColor);
+    }
+
+
```
Comment (mhabzda) [inline src/main/java/monitoringsystemturbo/presenter/MotivesPresenter.java]: same here

**Review by tomsoonn — APPROVED** [review]
a little to many empty lines, but seems ok

**Review by mhabzda — APPROVED** [review]

**Pull request merged**